In [1]:
import pandas as pd
import numpy as np

# Import FIle

In [2]:
df = pd.read_csv (r'C:\2020-XTern-DS.csv')

In [3]:
df.head(2)

Restaurant   Latitude  Longitude                                Cuisines  \
0    ID_6321  39.262605 -85.837372  Fast Food, Rolls, Burger, Salad, Wraps   
1    ID_2882  39.775933 -85.740581                     Ice Cream, Desserts   

  Average_Cost Minimum_Order Rating Votes Reviews   Cook_Time  
0       $20.00        $50.00    3.5    12       4  30 minutes  
1       $10.00        $50.00    3.5    11       4  30 minutes

# Fix / Normalize Data

In [4]:
df.shape

(2019, 10)

Remove any records where Average_Cost column has non-currency values

In [5]:
df = df[df.Average_Cost.str[:1] == '$']

Convert Average_Cost to numeric 

In [6]:
df ['Average_Cost'] = np.float32(df.Average_Cost.str.replace('$',''))

In [7]:
df.shape

(2017, 10)

Convert Reviews column to numeric

In [8]:
df['Reviews'] = df.Reviews.str.replace('-','0')

In [9]:
df['Reviews'] = np.int32(df.Reviews)

# Insights

### Insight #1

In [10]:
df[df.Average_Cost==np.max(df.Average_Cost)]

Restaurant   Latitude  Longitude             Cuisines  Average_Cost  \
2018    ID_8403  39.460944 -85.484355  Desserts, Beverages         150.0   

     Minimum_Order Rating Votes  Reviews   Cook_Time  
2018        $50.00    4.1   271      172  45 minutes

> Highest Average cost is $150 at Restaurant ID_8403. From this, we can extrapolate that this restaurant location has, on average, the most expensive menu in our dataset.

### Insight #2

In [11]:
np.min(df.Average_Cost)

5.0

In [12]:
len(df[df.Average_Cost==np.min(df.Average_Cost)])

49

In [13]:
df[df.Average_Cost==np.min(df.Average_Cost)]

Restaurant   Latitude  Longitude  \
13       ID_354  39.691918 -85.441037   
89      ID_2376  39.829000 -85.931205   
127     ID_5969  39.694712 -85.994750   
134     ID_4064  39.027065 -85.464734   
145     ID_6951  39.735048 -85.635146   
179     ID_2002  39.855502 -85.184145   
230     ID_6002  39.760450 -85.432839   
292     ID_7271  39.399979 -85.019125   
332     ID_4881  39.991266 -85.660687   
369     ID_5461  39.124551 -85.043515   
379     ID_7080  39.680171 -85.098364   
431     ID_6486  39.530358 -85.477282   
489     ID_3673  39.286978 -85.543741   
566     ID_3768  39.798598 -85.279668   
595     ID_4301  39.527469 -85.996286   
638     ID_8549  39.164010 -85.863011   
647     ID_2133  39.183237 -85.659096   
676     ID_2376  39.862749 -85.629410   
701      ID_635  39.748078 -85.735573   
796      ID_345  39.402880 -85.374808   
800     ID_7271  39.181314 -85.799288   
802     ID_1769  39.277783 -85.238398   
814     ID_1004  39.704981 -85.426350   
845      ID_629  39.533150 -85.651724   
877     ID_7239  39.205428 -85.273131   
1004     ID_781  39.350606 -85.406321   
1032    ID_1398  39.117908 -85.320079   
1080    ID_4824  39.347050 -85.754337   
1100    ID_2866  39.411517 -85.997416   
1182    ID_7271  39.397313 -85.570538   
1184     ID_781  39.463766 -85.027040   
1221    ID_7535  39.523482 -85.844197   
1261     ID_635  39.446849 -85.027294   
1284    ID_5435  39.302817 -85.930884   
1334    ID_7797  39.678473 -85.386408   
1358     ID_148  39.518481 -85.455297   
1367    ID_2370  39.613239 -85.180588   
1431    ID_4267  39.775161 -85.284794   
1513    ID_2005  39.484987 -85.135316   
1554    ID_5129  39.336366 -85.811928   
1567     ID_996  39.922521 -85.160773   
1580    ID_3798  39.730974 -85.382602   
1660    ID_3152  39.987348 -85.563654   
1770    ID_3833  39.014968 -85.516382   
1824    ID_7077  39.894889 -85.535273   
1844    ID_8287  39.616420 -85.106993   
1931    ID_4787  39.211529 -85.528237   
1942    ID_1517  39.346581 -85.284779   
1978    ID_5474  39.572524 -85.023807   

                                               Cuisines  Average_Cost  \
13                                            Beverages           5.0   
89                                         South Indian           5.0   
127                                              Mithai           5.0   
134                                         Street Food           5.0   
145                                        South Indian           5.0   
179                                               Rolls           5.0   
230                                              Bakery           5.0   
292                                        South Indian           5.0   
332                                              Juices           5.0   
369                                                Paan           5.0   
379                                   Juices, Beverages           5.0   
431                                Fast Food, Beverages           5.0   
489                                         Street Food           5.0   
566                                         Street Food           5.0   
595                                                Paan           5.0   
638                                         Street Food           5.0   
647                                        South Indian           5.0   
676                                        South Indian           5.0   
701                                              Mishti           5.0   
796                                        North Indian           5.0   
800                                        South Indian           5.0   
802                                         Street Food           5.0   
814                                         Street Food           5.0   
845   Street Food, Fast Food, Beverages, Desserts, M...           5.0   
877                                        North Indian           5.0   
1004                                Mithai, Street Food 

In [14]:
df[df.Average_Cost==np.min(df.Average_Cost)].Cuisines.value_counts()

South Indian                                           10
Street Food                                             9
Paan                                                    3
Mishti                                                  3
North Indian                                            3
Desserts                                                2
Beverages                                               2
Juices                                                  2
Mithai, Street Food                                     2
Fast Food, Beverages                                    2
Rolls                                                   1
North Indian, Chinese, Fast Food                        1
North Indian, Fast Food                                 1
Street Food, South Indian                               1
Fast Food                                               1
Bakery                                                  1
Mithai                                                  1
Momos, Fast Fo

> 49 restaurants are tied for the "Lowest Average Cost" at $5.00 The most promenent cuisines at this price point are *South Indian* followed by *Street Food*.

### Insight #3

In [15]:
df.head(1)

Restaurant   Latitude  Longitude                                Cuisines  \
0    ID_6321  39.262605 -85.837372  Fast Food, Rolls, Burger, Salad, Wraps   

   Average_Cost Minimum_Order Rating Votes  Reviews   Cook_Time  
0          20.0        $50.00    3.5    12        4  30 minutes

In [16]:
df[df.Reviews == np.max(df.Reviews)]

Restaurant   Latitude  Longitude                               Cuisines  \
1180    ID_1064  39.587777 -85.564805  Continental, North Indian, BBQ, Steak   

      Average_Cost Minimum_Order Rating Votes  Reviews   Cook_Time  
1180          80.0        $50.00    4.7  9054     6504  45 minutes

> Based on the number of reviews, Restaurant ID_1064 appears to be the most popular.

### Insight #4

In [17]:
df.head(1)

Restaurant   Latitude  Longitude                                Cuisines  \
0    ID_6321  39.262605 -85.837372  Fast Food, Rolls, Burger, Salad, Wraps   

   Average_Cost Minimum_Order Rating Votes  Reviews   Cook_Time  
0          20.0        $50.00    3.5    12        4  30 minutes

In [18]:
nonzero_rating = df[(df.Reviews>0)]

In [19]:
nonzero_rating[nonzero_rating.Rating == np.min(nonzero_rating.Rating)]

Restaurant   Latitude  Longitude      Cuisines  Average_Cost  \
287    ID_5908  39.221608 -85.030116  North Indian          20.0   

    Minimum_Order Rating Votes  Reviews   Cook_Time  
287        $50.00    2.4    19       11  30 minutes

> Restaurant ID_5908 is the lowest rated restaurant that has at least one review, implying it's the least favorable restaurant location in our dataset.